src/main/java/com/targetindia/DigitizeTimeTable/controller/AdminController.java

```diff
@@ -0,0 +1,132 @@
+
+package com.targetindia.DigitizeTimeTable.controller;
+
+import com.targetindia.DigitizeTimeTable.model.ClassInfo;
+import com.targetindia.DigitizeTimeTable.model.Instructor;
+import com.targetindia.DigitizeTimeTable.model.Course;
+import com.targetindia.DigitizeTimeTable.service.AdminService;
+//import org.mockito.internal.matchers.Null;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("admins")
+public class AdminController {
```

Comment:
```
@RestController
@RequestMapping("admins")
public class AdminController {
    private static final Logger logger = LoggerFactory.getLogger(AdminController.class);

    private final AdminService service;

    @Autowired
    public AdminController(AdminService service) {
        this.service = service;
    }

    @PostMapping("/instructors")
    public ResponseEntity<String> addInstructor(@RequestBody Instructor instructor) {
        service.addInstructor(instructor);
        return new ResponseEntity<>("Instructor added successfully", HttpStatus.CREATED);
    }
```

Reply:
and rest of the methods. Make use of Constructor-based Dependency Injection rather than field-based. This is recommended as it helps in writing testable code.